src/app_service/service/wallet_account/service.nim

```diff
@@ -437,35 +485,36 @@ QtObject:
     try:
       let responseObj = response.parseJson
       var data = TokensPerAccountArgs()
-      let walletAddresses = toSeq(self.walletAccounts.keys)
-      for wAddress in walletAddresses:
-        var tokensArr: JsonNode
-        var tokens: seq[WalletTokenDto]
-        if(responseObj.getProp(wAddress, tokensArr)):
-          tokens = map(tokensArr.getElems(), proc(x: JsonNode): WalletTokenDto = x.toWalletTokenDto())
-        
-          tokens.sort(priorityTokenCmp)
-          self.walletAccounts[wAddress].tokens = tokens
-          data.accountsTokens[wAddress] = tokens
-          self.tokenService.updateTokenPrices(tokens) # For efficiency. Will be removed when token info fetching gets moved to the tokenService
+      if responseObj.kind == JObject:
+        for wAddress, tokensDetailsObj in responseObj:
+          if tokensDetailsObj.kind == JArray:
+            var tokens: seq[WalletTokenDto]
+            tokens = map(tokensDetailsObj.getElems(), proc(x: JsonNode): WalletTokenDto = x.toWalletTokenDto())
+            tokens.sort(priorityTokenCmp)
+            data.accountsTokens[wAddress] = tokens
+            self.storeTokensForAccount(wAddress, tokens)
+            self.tokenService.updateTokenPrices(tokens) # For efficiency. Will be removed when token info fetching gets moved to the tokenService
       self.events.emit(SIGNAL_WALLET_ACCOUNT_TOKENS_REBUILT, data)
     except Exception as e:
       error "error: ", procName="onAllTokensBuilt", errName = e.name, errDesription = e.msg
 
-  proc buildAllTokens(self: Service, accounts: seq[string]) =
+  proc buildAllTokens(self: Service, accounts: seq[string] = @[]) =
```

Comment:
@dlipicar see the implementation of `buildAllTokens`

Reply:
I was thinking of some scenario like `self.buildAllTokens(self.getAddressesOfWatchOnlyAccounts())`. If the user has no watch-only accounts, with the new implementation all tokens will be rebuilt instead of doing nothing.

Reply:
@dlipicar yes, that's something we don't have as of now in code, but it may be a need for something like that in future. Anyhow I updated the code and added the check, which was not there before, to ignore building tokens if the provided accounts list is empty.